src/naive/internals.rs

```diff
@@ -322,14 +329,19 @@ impl Mdf {
     ///
     /// # Errors
     ///
-    /// Returns `None` if `month == 0` or `day == 0`, or if a the given day does not exist in the
-    /// given month.
+    /// Returns [`Error::InvalidArgument`] if `month == 0` or `day == 0`.
+    ///
+    /// Returns [`Error::DoesNotExist`] if the given day does not exist in the given month.
     #[inline]
-    pub(super) const fn ordinal(&self) -> Option<u32> {
+    pub(super) const fn ordinal(&self) -> Result<u32, Error> {
         let mdl = self.0 >> 3;
-        match MDL_TO_OL[mdl as usize] {
-            XX => None,
-            v => Some((mdl - v as u8 as u32) >> 1),
+        let adjustment = MDL_TO_OL[mdl as usize];
```

Comment:
Would it be possible to keep the matching? That way the adjustment variable has a tighter scope.

Secondly `x < 0` is not always the same as `x == -1`. Currently there is no other value than `-1`, but it is slightly more future proof to immediately detect `-2` as a different error. A more suitable error would probably be `Error::Undefined`, but it is out of range in this case.
```
        match MDL_TO_OL[mdl as usize] {
            adjustment if adjustment > 0 => Ok((mdl - adjustment as u8 as u32) >> 1),
            adjustment if adjustment == 0 => Err(Error::DoesNotExist),
            adjustment if adjustment == 1 => Err(Error::InvalidArgument),
            _ => Err(Error::OutOfRange),
        }
```

Reply:
My first attempt used `cmp(&0)`, but that didn't work in const context. I expect making the match more precise instead of positive/0/negative would effect the performance a bit. (Can't run a reliable benchmark on the device I am working on right now.)

Reply:
Very nice!